shaders/src/main.fs

```diff
@@ -28,6 +28,17 @@ void main() {
 
     fragColor = evaluateMaterial(inputs);
 
+#if defined(HAS_FOG)
+    // we need the real world position, which is obtained by adding the world-offset
+    vec3 wsCameraPostion = getWorldCameraPosition() + getWorldOffset();
+    vec3 wsFragPosition = getWorldPosition() + getWorldOffset();
+
+    // now we need the world-space vector from camera to the shaded fragment
+    vec3 cameraToWorld = wsFragPosition - wsCameraPostion;
```

Comment:
```suggestion
    vec3 cameraToWorld = worldPosition - worldCameraPostion;
```

Reply:
`worldPosition` is confusing to me. It's the position of what?
`wsFragPosition` is clear "fragment position expressed is world-space".
I prefer my convention :-)

Reply:
Sure, but it's the convention already in use in the shaders :p

worldPosition is the position in world space, it's clear :) When there's no descriptor it entails the position of the current point (a fragment doesn't make sens in world space :p).

Reply:
okay I can rename to wsFragCoord :-)  More seriously, I don't like it when naming has conditions in them. Like "If there is no descriptor, then it means X". this could be dependent on context, and it means you need to read the doc (which we don't have) before you can read the code. I prefer when all names have the same format.

Reply:
You can keep "frag" if you want.